### ISSUE TITLE
Setup moveit2 simulation

### ISSUE THREAD

**Issue by dylan-gonzalez** [MonashNovaRover/nova#64]
**CONTEXT**
Basic simulation with an arm URDF off of the internet, or our current arm URDF.

**TODO**
- Setup moveit2 simulation

  
  Tags for Notion Integration:
  tag:arm;tag:software

**Comment by RohitPilakkat** [comment]
I have tried to get moveit2 onto the workspace but was running into some issues with a particular dependency not cloning when installing moveit through nix-ros-overlay. I left an issue on their repo and hopefully I'll figure it out from there.

https://github.com/lopsided98/nix-ros-overlay/issues/473#issuecomment-2355462730

**Comment by dylan-gonzalez** [comment]
The comment there is correct

The [git clone](https://github.com/tier4/osqp_vendor/blob/1ea3682dacc08bd05d301098b202e74bd6416251/CMakeLists.txt#L22-L33) won't work since nix sandboxes its builds, so there is no internet access during the build phase.

I had a similar issue and have a pending [PR](https://github.com/MonashNovaRover/nixfiles/pull/65) that resolves it by grabbing the files to be downloaded in the fetch phase.

**Comment by RohitPilakkat** [comment]
I think I've tried running the environment with `--option sandbox false` before but that didn't work. Not entirely sure what that option does when turning off sandboxing. I'll have a look at that PR you made and see if I can adapt it to this though, thanks.

**Comment by dylan-gonzalez** [comment]
Try setting [this](https://mynixos.com/nixpkgs/option/nix.settings.sandbox) to `relaxed` ?

**Comment by RohitPilakkat** [comment]
That wouldn't be a long-term solution though would it? I'll try these and see what works best when I'm home later tonight though, thanks again.

**Comment by dylan-gonzalez** [comment]
> That wouldn't be a long-term solution though would it

no, so maybe it's feasible to do something like i did with the PR i linked, but haven't looked into the moveit2 package properly

**Comment by RohitPilakkat** [comment]
I'll have a go with it later tonight and hopefully it works well!

**Comment by RohitPilakkat** [comment]
https://github.com/MonashNovaRover/nixfiles/tree/feat/setup_osqp-vendor_moveit2/overlay/ros

Made a branch here to try and make it work. Still running into issues with what seems like nix' fixupPhase that I'm not sure how to work around yet.

**Comment by RohitPilakkat** [comment]
That branch contains a fix for the osqp-vendor issue. There's an additional issue with octomap not being able to access a certain configuration file version that I'll have to look into before moveit-ros is working (Assuming there are no other issues after).

**Comment by RohitPilakkat** [comment]
This is patched through maintenance.nix btw and moveit2 should be pullable through nix-ros-overlay now (has been since like Oct last year).

https://github.com/MonashNovaRover/nova/pulls?page=3&q=is:pr%20is:closed

**Comment by V01DBREAKER** [comment]
@BaileyChessum @RohitPilakkat Thoughts now post URC?
